File: IT/Stringhe/Esercizi/Java/Esercizio23.java
/* file Esercizio23.java */

import java.util.regex.Pattern;
import java.util.regex.Matcher;

public class Esercizio23 {
	public static void main(String [] args) {
		String s = "qui. quo. qua.";

		System.out.println(s);

		Pattern pattern = Pattern.compile("(^[\\w]|\\.\\s[\\w])");
		Matcher matcher = pattern.matcher(s);
		StringBuffer sb = new StringBuffer();

		while (matcher.find()) {
			matcher.appendReplacement(sb, matcher.group(1).toUpperCase());
		}

		matcher.appendTail(sb);

		s = sb.toString();
		System.out.println(s);
	}
}

File: IT/Stringhe/Esercizi/Python/esercizio23.py
# file esercizio23.py

from __future__ import print_function, unicode_literals
import re

s = "qui. quo. qua."

print(s)

s = re.sub(r"(^[\w]|\.\s[\w])", lambda m : m.group(1).upper(), s)

print(s)

